library/src/scala/tasty/reflect/PrinterOps.scala

```diff
@@ -0,0 +1,90 @@
+package scala.tasty
+package reflect
+
+import scala.quoted.show.SyntaxHighlight
+
+trait PrinterOps extends Core { self: Reflection =>
+
+  /** Adds `show` as an extension method of a `Tree` */
+  implicit class TreeShowDeco(tree: Tree) {
+    /** Shows the tree as extractors */
+    def showExtractors(given ctx: Context): String =
+      val printers = new Printers(self)
+      new printers.ExtractorsPrinter().showTree(tree)
+
+    /** Shows the tree as fully typed source code */
+    def show(given ctx: Context): String = show(SyntaxHighlight.plain)
+
+    /** Shows the tree as fully typed source code */
+    def show(syntaxHighlight: SyntaxHighlight)(given ctx: Context): String =
+      val printers = new Printers(self)
+      new printers.SourceCodePrinter(syntaxHighlight).showTree(tree)
+
+  }
```

Comment:
Use extension methods here (and below), instead of implicit class?

Reply:
Those are blocked by #7401